code/modules/urist/modules/shipbattles/ship_overmaps.dm

```diff
@@ -273,4 +268,7 @@
 
 /obj/effect/overmap/visitable/ship/combat/proc/assign_target_zs() //this is a proc so it can be overridden by non-nerva ships. restricting zs from being hit for awaymaps is a little more complicated because they don't have set z-levels, but can be done here if you get creative.
 	if(!target_zs)
-		target_zs = map_z
\ No newline at end of file
+		target_zs = map_z
+
+/obj/effect/overmap/visitable/ship/combat/proc/pve_mapfire(var/projectile_type)
```

Comment:
Seems odd to me. I _think_ what you intended to do was make `pve_mapfire` a proc of `/mob/living/simple_animal/hostile/overmapship`? That way each template can handle how projectiles come in etc? Doesn't really make sense for different player ships to need this function

Reply:
This is intentional so each combat ship can define where they want hostile ships spawned in. This way you could add a blank z level to a new combat ship awaymap as the last map loaded, and then use max(map_z) to fire projectiles there. Boarding code would need a slight rework to make ships spawn there properly, but that's for a future PR.

Reply:
I think I'm getting confused between the current state and the WIP rework I was playing around with that allowed fr multiple ship templates spawning in over multiple Zs. Been a long time since I looked at this code XD
Think there's definitely a followup PR somewhere in all this

Reply:
👀 that sounds very interesting